example/lib/model/ble_device.dart

```diff
@@ -1,89 +1,48 @@
-import 'dart:async';
-
 import 'package:collection/collection.dart';
 import 'package:flutter_ble_lib/flutter_ble_lib.dart';
 
-abstract class BleDevice {
-  String id;
-  int counter = 0;
+class BleDevice {
+  final Peripheral peripheral;
   final String name;
-  DeviceCategory _category;
-  bool _isConnected = false;
-  Peripheral peripheral;
-
-  bool get isConnected =>
-      _isConnected;
-
-  DeviceCategory get category => _category;
-
-  BleDevice(this.name, this.id, this.peripheral) {
-    _category = _nameToCategory(name);
-    if(this.name == null) {
-
-    }
-  }
+  final DeviceCategory category;
 
-  factory BleDevice.connected(BleDevice bleDevice) {
-    return ConnectedBleDevice(bleDevice.name, bleDevice.id, bleDevice.peripheral);
-  }
+  String get id => peripheral.identifier;
 
-  factory BleDevice.notConnected(String name, String id, Peripheral peripheral) {
-    return DisconnectedBleDevice(name, id, peripheral);
-  }
-
-  DeviceCategory _nameToCategory(String name) {
-    if (name == "SensorTag") {
-      return DeviceCategory.sensorTag;
-    } else if (name!= null && name.startsWith("Hex")) {
-      return DeviceCategory.hex;
-    } else {
-      return DeviceCategory.other;
-    }
-  }
+  BleDevice(ScanResult scanResult)
+      : peripheral = scanResult.peripheral,
+        name = scanResult.name,
+        category = scanResult.category;
 
   @override
-  int get hashCode => 123;
+  int get hashCode => id.hashCode;
 
   @override
   bool operator ==(other) =>
-      other is BleDevice
-          && this.name != null
-          && other.name != null
-          && compareAsciiLowerCase(this.name, other.name) == 0
-          && this.id == other.id;
+      other is BleDevice &&
```

Comment:
Why the change? I think it was quicker to read with connecting operators on the beginning of the line.

Reply:
I didn't change it manually. Auto-format on save. 😄